src/aspire/denoising/denoiser_cov2d.py

```diff
@@ -0,0 +1,79 @@
+import logging
+import numpy as np
+
+from aspire.denoising import Denoiser
+from aspire.utils.filters import RadialCTFFilter
+from aspire.utils.blk_diag_func import blk_diag_partition
+from aspire.utils.blk_diag_func import blk_diag_eye
+from aspire.basis.ffb_2d import FFBBasis2D
+from aspire.estimation.covar2d import RotCov2D
+
+logger = logging.getLogger(__name__)
+
+
+class DenoiserCov2D(Denoiser):
+    """
+    Define a derived class for denoising 2D images using Cov2D method
+    """
+    def __init__(self, src, basis, var_noise, ctf_info):
+        """
+        Initialize an object for denoising 2D images using Cov2D method
+
+        :param src: The source object of 2D images with metadata
+        :param basis: The basis method to expand 2D images
+        :param var_noise: The estimated variance of noise
+        :param ctf_info: Whether the CTF filters are included
+        """
+        super().__init__(src)
+        self.imgs_noise = self.src.images(start=0, num=np.inf).asnumpy()
+        self.var_noise = var_noise
+        if not isinstance(basis, FFBBasis2D):
```

Comment:
Regular `FBBasis2D` doesn't work?

Reply:
I did not go thought the regular FB basis. Need to check it for latest version.

Reply:
Opened the issue #106 for future implementation.